scripts/source-maintenance.sh

```diff
@@ -496,6 +559,14 @@ run_ checkMakeNamedErrorDetails || exit 1
 # Run formatting
 srcFormat || exit 1
 
+if [ -z "$OnlyChangedSince" ]; then
+    sort -u <doc/debug-sections.tmp | sort -n >doc/debug-sections.tmp2
+    cat scripts/boilerplate.h doc/debug-sections.tmp2 >doc/debug-sections.txt
+    rm doc/debug-sections.tmp2
+else
+    echo "--only-changed-since specified, Skipping update of doc/debug-sections.txt"
+fi
+rm -f doc/debug-sections.tmp
```

Comment:
I do not see how the above badly duplicated code can work together with `--only-changed-since`: By the time we get here, processDebugSections() were already executed and deleted `doc/debug-sections.tmp` AFAICT. Please move this `--only-changed-since` exclusion logic into processDebugSections() _and test_!

```suggestion
```